src/chainparams.cpp

```diff
@@ -249,13 +254,95 @@ class CTestNetParams : public CChainParams {
     }
 };
 
+/**
+ * Signet
+ */
+class SigNetParams : public CChainParams {
+public:
+    explicit SigNetParams(const ArgsManager& args) {
+        std::vector<uint8_t> bin;
+        vSeeds.clear();
+
+        if (!args.IsArgSet("-signet_blockscript")) {
+            LogPrintf("Using default signet network\n");
+            bin = ParseHex("512103ad5e0edad18cb1f0fc0d28a3d4f1f3e445640337489abb10404f2d1e086be43051ae");
+            vSeeds.emplace_back("178.128.221.177");
+            vSeeds.emplace_back("2a01:7c8:d005:390::5");
+            vSeeds.emplace_back("ntv3mtqw5wt63red.onion:38333");
+        } else {
+            const auto blockscript = args.GetArgs("-signet_blockscript");
+            if (blockscript.size() != 1) {
+                throw std::runtime_error(strprintf("%s: -signet_blockscript cannot be multiple values.", __func__));
+            }
+            bin = ParseHex(blockscript[0]);
+
+            LogPrintf("Signet with block script %s\n", blockscript[0]);
```

Comment:
Do you think it would be helpful to print the decoded ASM of the script as well?

Reply:
From personal experience, this is a check point to ensure you're not accidentally running on the wrong signet, in which case the ASM won't really help any.